contracts/strategies/RFPStrategy.sol

```diff
@@ -0,0 +1,278 @@
+    }
+
+    /// @notice Get the status of the milestone
+    /// @param _milestoneId Id of the milestone
+    function getMilestoneStatus(uint256 _milestoneId) public view returns (RecipientStatus) {
+        return milestones[_milestoneId].milestoneStatus;
+    }
+
+    /// ===============================
+    /// ======= External/Custom =======
+    /// ===============================
+
+    /// @notice Set milestones for RFP pool
+    /// @param _milestones The milestones to be set
+    function setMilestones(Milestone[] memory _milestones) external onlyPoolManager(msg.sender) {
+        uint256 milestonesLength = _milestones.length;
+        for (uint256 i = 0; i < milestonesLength;) {
+            milestones.push(_milestones[i]);
+
+            unchecked {
+                i++;
+            }
+        }
+
+        emit MILESTONES_SET();
+    }
+
+    /// @notice Submit milestone to RFP pool
+    /// @param _metadata The proof of work
+    function submitMilestone(Metadata calldata _metadata) external {
+        if (acceptedRecipientId != msg.sender && !_isIdentityManager(acceptedRecipientId, msg.sender)) {
+            revert UNAUTHORIZED();
+        }
+
+        if (upcomingMilestone > milestones.length) {
+            revert INVALID_MILESTONE();
+        }
+
+        milestones[upcomingMilestone].metadata = _metadata;
+        milestones[upcomingMilestone].milestoneStatus = RecipientStatus.Pending;
+
+        emit MILESTONE_SUBMITTED(upcomingMilestone);
+    }
+
+    /// @notice Update max bid for RFP pool
+    /// @param _maxBid The max bid to be set
+    function updateMaxBid(uint256 _maxBid) external onlyPoolManager(msg.sender) {
+        maxBid = _maxBid;
+
+        emit MAX_BID_UPDATED(maxBid);
+    }
+
+    /// @notice Reject pending milestone
+    /// @param _milestoneId Id of the milestone
+    function rejectMilestone(uint256 _milestoneId) external onlyPoolManager(msg.sender) {
+        if (milestones[_milestoneId].milestoneStatus == RecipientStatus.Accepted) {
+            revert MILESTONE_ALREADY_ACCEPTED();
+        }
+
+        milestones[_milestoneId].milestoneStatus = RecipientStatus.Rejected;
+        emit MILESTONE_REJECTED(_milestoneId);
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @notice Submit proposal to RFP pool
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the transaction
+    function _registerRecipient(bytes memory _data, address _sender) internal override returns (address recipientId) {
+        if (acceptedRecipientId != address(0)) {
+            revert RECIPIENT_ALREADY_ACCEPTED();
+        }
+
+        address recipientAddress;
+        uint256 proposalBid;
+        Metadata memory metadata;
+
+        // decode data custom to this strategy
+        if (registryGating) {
+            (recipientId, recipientAddress, proposalBid, metadata) =
```

Comment:
we cant use the anchor as identyId
since the anchor can change during the pool period

if the anchor changes during the pool period, this check on `submitMilestone` will fail.

to still be able to rely on the anchor we could maybe remove `anchorToIdentityId[identity.anchor] = bytes32(0);` on the `Registry` in `updateIdentityName`
-> we should carefully think about pros and cons

Reply:
@nfrgosselin @zobront @codenamejason ^

Reply:
I think it is a problem. Let's picture this situation:
- my identity id = 1 and name = Zach Obront => this outputs Anchor 1
- i change my name to Elon Musk => this outputs Anchor 2
- any attestation to Anchor 1 is still considered valid, which means I can submit to projects as Elon Musk (because the anchor is pointing to the new struct that has been updated)

***

I believe the current setup in this contract also limits that at most one recipient can be registered per `recipientId` / `anchor`, which may limit a situation when there are multiple members from one identity that want to submit.

It seems like the ideal solution is that, rather than tracking recipients by `recipientId`, we track them by `recipientAddress`. That means `recipientId` is only a blocker when submitting, to make sure the identity is valid, and after that it's just there for name/metadata. 

***

This still leaves the problem that the link would break if they changed the name (because that would also change the anchor). But we probably want the behavior, not to just assume the new name is valid? 

The only solution to that I can see is that you can never change a name in the Registry. It has to continue to exist forever. You can simply create a new identity in the Registry, with a new name, and then base things on that.

I actually think that may be better, as it ensures historical ones are always there, and doesn't seem to harm users at all.

Thoughts?

Reply:
> The only solution to that I can see is that you can never change a name in the Registry. It has to continue to exist forever. You can simply create a new identity in the Registry, with a new name, and then base things on that.
I actually think that may be better, as it ensures historical ones are always there, and doesn't seem to harm users at all.

This is actually where I thought we ended up. I'm okay with this, as we can normalize an attestation pattern that allows old identities to validate their new identities